app/models/assignment.rb

```diff
@@ -1464,4 +1464,68 @@ def same_membership_as_csv_row?(row, existing_grouping)
     end
   end
 
+  # Reformatted logic for download_assignment_list controller method
+  # Returns an output file for controller to handle.
+  def self.get_assignment_list(file_format, default_fields)
+    assignments = self.all
+
+    case file_format
+    when 'yml'
+      map = {}
+      map[:assignments] = []
+      assignments.map do |assignment|
+        m = {}
+        default_fields.length.times do |i|
+          m[default_fields[i]] = assignment.send(default_fields[i])
+        end
+        map[:assignments] << m
+      end
+      output = map.to_yaml
```

Comment:
Ruby's functions automatically return the last evaluated value.

This means that you can remove the `output` variable altogether, and simply write `map.to_yaml` here, and that's what will be returned. (Similarly in the "csv" branch, simply write `MarkusCSV.generate(assignments) do ...` and the value will be returned.